src/main/java/com/github/manolo8/darkbot/core/manager/HangarManager.java

```diff
@@ -0,0 +1,151 @@
+                try {
+                    backpageManager.getConnection(url).getResponseCode();
+                    this.disconnecting = false;
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+                this.lastChangeHangar = System.currentTimeMillis();
+            }
+        } else {
+            return false;
+        }
+
+        return true;
+    }
+
+    public void updateDrones() {
+        try {
+            String hangarID = getActiveHangar();
+
+            if (hangarID != null) {
+                String decodeParams = "{\"params\":{\"hi\":" + hangarID + "}}";
+                String encodeParams = Base64.getEncoder().encodeToString(decodeParams.getBytes("UTF-8"));
+                String url = "flashAPI/inventory.php?action=getHangar&params="+encodeParams;
+                String json = this.backpageManager.getDataInventory(url);
+                JsonArray hangarArray =  new JsonParser().parse(json).getAsJsonObject().get("data")
+                        .getAsJsonObject().get("ret").getAsJsonObject().get("hangars").getAsJsonArray();
+
+                for (JsonElement hangar : hangarArray) {
+                    if (hangar.getAsJsonObject().get("hangar_is_active").getAsBoolean()) {
+                        JsonArray dronesArray = hangar.getAsJsonObject().get("general").getAsJsonObject().get("drones").getAsJsonArray();
+                        for (JsonElement dron : dronesArray){
+                            JsonObject dronJson = dron.getAsJsonObject();
+                            String lootId = "drone_iris";
+                            switch (dronJson.get("L").getAsInt()){
+                                case 1:
+                                    lootId = "drone_flax";
+                                    break;
+                                case 2:
+                                    lootId = "drone_iris";
+                                    break;
+                                case 3:
+                                    lootId = "drone_apis";
+                                    break;
+                                case 4:
+                                    lootId = "drone_zeus";
+                                    break;
+                            }
+                            int damage = Integer.parseInt(dronJson.get("HP").getAsString().replace("%"," ").trim());
+                            this.drones.add(new Drone(lootId,dronJson.get("repair").getAsInt(),dronJson.get("I").getAsString(),
+                                    dronJson.get("currency").getAsString(),dronJson.get("LV").getAsInt(),damage));
+                        }
+                    }
+                }
+            }
+
+        } catch (Exception e){
+            e.printStackTrace();
+        }
+    }
+
+    private boolean repairDron(Drone drone, String activeHangarId){
+        try {
+            String decodeParams =
```

Comment:
make this a POJO, wich you convert to JSON via GSON, and then base64 enode it. Put such a POJO to String function in the same util class as the base64 string to POJO util, along the lines of:
`public static <T> String encodeRequest(T obj)`